backends/aoti/runtime/AotiBackend.cpp

```diff
@@ -0,0 +1,532 @@
+
+#include <executorch/backends/arm/runtime/VelaBinStream.h>
+#include <executorch/extension/tensor/tensor.h>
+#include <executorch/runtime/backend/interface.h>
+#include <executorch/runtime/core/error.h>
+#include <executorch/runtime/core/evalue.h>
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include "cuda_runtime.h"
+
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace executorch {
+namespace backends {
+namespace aoti {
+
+// Here is where the aoti bouncers are going to be defined.
+// I define the globals aoti generated compiled code calls
+// They can be backed by ET systems
+
+using namespace std;
+
+using executorch::aten::ScalarType;
+using executorch::runtime::ArrayRef;
+using executorch::runtime::Backend;
+using executorch::runtime::BackendExecutionContext;
+using executorch::runtime::BackendInitContext;
+using executorch::runtime::CompileSpec;
+using executorch::runtime::DelegateHandle;
+using executorch::runtime::Error;
+using executorch::runtime::EValue;
+using executorch::runtime::FreeableBuffer;
+using executorch::runtime::MemoryAllocator;
+using executorch::runtime::Result;
+using executorch::runtime::etensor::Tensor;
+
+
+extern "C" {
+  using AOTITensorHandle = Tensor*;
+
+  // TODO: We should get a proper one
+  struct CUDAStreamGuardOpaque;
+  using CUDAStreamGuardHandle = CUDAStreamGuardOpaque*;
+
+  using AOTIRuntimeError = Error;
+  using AOTITorchError = Error;
+
+  struct AOTInductorModelContainerOpaque;
+  using AOTInductorModelContainerHandle = AOTInductorModelContainerOpaque*;
+  using AOTInductorStreamHandle = void*;
+  using AOTIProxyExecutorHandle = void*;
+
+  using AOTInductorModelContainerCreateWithDeviceFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle* container_handle,
+    size_t num_models,
+    const char* device_str,
+    const char* cubin_dir);
+
+  using AOTInductorModelContainerDeleteFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle container_handle);
+
+  using AOTInductorModelContainerGetNumInputsFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle container_handle,
+    size_t* num_constants);
+
+  using AOTInductorModelContainerGetNumOutputsFunc = AOTIRuntimeError(*)(
+    AOTInductorModelContainerHandle container_handle,
+    size_t* num_constants);
+
+  using AOTInductorModelContainerRunFunc = AOTIRuntimeError(*)(
```

Comment:
this stuff looks like it should go with (or is copy/pasted from, but I can't find it) AOTInductorModelContainer. why is it here?